src/boot/shim.c

```diff
@@ -33,12 +33,21 @@ struct ShimLock {
 #define SHIM_LOCK_GUID \
         { 0x605dab50, 0xe046, 0x4300, { 0xab, 0xb6, 0x3d, 0xd8, 0x10, 0xdd, 0x8b, 0x23 } }
 
+#define SHIM_IMAGE_LOADER_GUID \
+        { 0x1f492041, 0xfadb, 0x4e59, { 0x9e, 0x57, 0x7c, 0xaf, 0xe7, 0x3a, 0x55, 0xab } }
+
 bool shim_loaded(void) {
         struct ShimLock *shim_lock;
 
         return BS->LocateProtocol(MAKE_GUID_PTR(SHIM_LOCK), NULL, (void **) &shim_lock) == EFI_SUCCESS;
 }
 
+bool shim_loader_available(void) {
+        void *shim_image_loader;
+
```

Comment:
please add a comment here that clarifies what this is about and why we check it.

Reply:
i.e. anyone who hasn't drunk the shim cool-aid (i.e. me) is very confused what the shim lock protocol is, and what the shim image loader thing is, and why we we need to check both

Reply:
ok, done